### PR TITLE
[bitnami/fluentd] Add ServiceAccount annotations

### LINKED ISSUE #3389: [bitnami/fluentd] Add ServiceAccount annotations
**Which chart**:
fluentd 1.2.11

**Is your feature request related to a problem? Please describe.**
Among others the external-dns chart supports configurable annotations on the ServiceAccounts, the fluentd chart doesn't.

**Describe the solution you'd like**
I'd like to configure ServiceAccount annotations the same way as in the external-dns chart.

**Describe alternatives you've considered**
Considered setting `.Values.serviceAccount.create` to `false`. That would require me managing the ServiceAccount, the ClusterRole and the ClusterRoleBinding outside of Helm.

**Additional context**
None

### PR DESCRIPTION
**Pull request by farkasmate**
**Description of the change**

Add ServiceAccount annotation support for fluentd chart.

**Benefits**

This would allow us to manage `eks.amazonaws.com/role-arn` annotations in the same way as in other Bitnami charts.

**Possible drawbacks**

None

**Applicable issues**

  - fixes #3389

**Additional information**

The implementation is a copy of the `bitnami/external-dns` feature.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/fluentd/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -4,4 +4,7 @@ kind: ServiceAccount
 metadata:
   name: {{ include "fluentd.serviceAccountName" . }}
   labels: {{- include "fluentd.labels" . | nindent 4 }}
-{{- end -}}
\ No newline at end of file
+  {{- if .Values.serviceAccount.annotations }}
+  annotations: {{ toYaml .Values.serviceAccount.annotations | nindent 4 }}
```
Comment (dani8art) [inline bitnami/fluentd/templates/serviceaccount.yaml:8]: Could we follow this approach using `fluentd.tplValue`?

https://github.com/bitnami/charts/blob/master/bitnami/fluentd/templates/aggregator-statefulset.yaml#L23

File: bitnami/fluentd/README.md [hunk 2]
```diff
@@ -140,6 +140,7 @@ The following tables lists the configurable parameters of the fluentd chart and
 | `aggregator.podAnnotations`                     | Pod annotations                                                                                                | `{}`                                                                                                    |
 | `serviceAccount.create`                         | Specify whether a ServiceAccount should be created                                                             | `true`                                                                                                  |
 | `serviceAccount.name`                           | The name of the ServiceAccount to create                                                                       | Generated using the `fluentd.fullname` template                                                         |
+| `serviceAccount.annotations`                    | Additional Service Account annotations                                                                         | `{}`                                                                                                    |
```
Comment (dani8art) [inline bitnami/fluentd/README.md:143]:
```suggestion
| `serviceAccount.annotations`                    | Additional Service Account annotations (evaluated as a template)                                                                       | `{}`                                                                                                    |
```

File: bitnami/fluentd/values-production.yaml [hunk 3]
```diff
@@ -407,6 +407,9 @@ serviceAccount:
   ## The name of the ServiceAccount to use.
   ## If not set and create is true, a name is generated using the fluentd.fullname template
   # name:
+  ## Annotations for the Service Account
```
Comment (dani8art) [inline bitnami/fluentd/values-production.yaml:410]:
```suggestion
  ## Annotations for the Service Account (evaluated as a template)
```

File: bitnami/fluentd/values.yaml [hunk 4]
```diff
@@ -407,6 +407,9 @@ serviceAccount:
   ## The name of the ServiceAccount to use.
   ## If not set and create is true, a name is generated using the fluentd.fullname template
   # name:
+  ## Annotations for the Service Account
```
Comment (dani8art) [inline bitnami/fluentd/values.yaml:410]:
```suggestion
  ## Annotations for the Service Account (evaluated as a template)
```

**Comment by farkasmate** [comment]
> Hi @farkasmate thank you so much for the PR
> 
> Please, take a look at my comments.

Hi @dani8art,

I look into it and update the PR.

**Comment by farkasmate** [comment]
Thank you.

**Pull request merged**